rts/lra/lra-coordinator/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java

```diff
@@ -149,7 +149,9 @@ public void finished(Transaction transaction, boolean fromHierarchy) {
             // the LRA is top level or it's a nested LRA that was closed by a
             // parent LRA (ie when fromHierarchy is true) then it's okay to forget about the LRA
 
-            remove(ActionStatus.stringForm(transaction.status()), transaction.getId());
+            if (transaction.afterLRANotification()) {
+                remove(ActionStatus.stringForm(transaction.status()), transaction.getId());
+            }
```

Comment:
sorry, but will this be called repeatedly? I don't see it from a quick look in the code. In other case, the specification says we should log a warning when we stop retrying - https://github.com/eclipse/microprofile-lra/blob/master/api/src/main/java/org/eclipse/microprofile/lra/annotation/AfterLRA.java#L62

Reply:
I think as a SHOULD we don't actually have to do it, but I agree it would be useful to add

Reply:
I created issue https://github.com/eclipse/microprofile-lra/issues/190